### PR TITLE
Clean up automap.cpp

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

**Comment by glebm** [comment]
Nice! Does this fix the map getting cut off at the bottom of the screen in widescreen?

image

File: Source/control.cpp [hunk 1]
```diff
@@ -981,6 +981,11 @@ void DoAutoMap()
 	}
 }
 
+bool CanPanelsCoverView()
```
Comment (glebm) [inline Source/control.cpp:984]: nit: best place this in the header to let the compiler inline this easily

functions in the header must be marked `inline`

**Comment by AJenbo** [comment]
unfortunately not, this is purl code cleanup with no change in behavior. But it shoud make it easier to fix the issue.

(it does fix a bug from the previous cleanup, and hide the game name if unknown (seamed to confuse a lot of players)

File: Source/automap.cpp [hunk 2]
```diff
@@ -23,7 +23,7 @@ namespace {
 /**
  * Maps from tile_id to automap type.
  */
-uint16_t AutomapTypes[256];
+std::array<uint16_t, 256> AutomapTypes = {};
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/automap.cpp:26]: ok right, i remembered it as the other way around

File: Source/automap.cpp [hunk 3]
```diff
@@ -379,36 +380,61 @@ uint16_t GetAutomapType(Point map, bool view)
...
-
 	char desc[256];
 	int nextLine = 20;
 
 	if (gbIsMultiplayer) {
-		strcat(strcpy(desc, _("game: ")), szPlayerName);
-		PrintGameStr(out, 8, 20, desc, COL_GOLD);
-		nextLine = 35;
-		if (szPlayerDescript[0] != 0) {
+		if (strcasecmp("0.0.0.0", szPlayerName) != 0) {
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/automap.cpp:387]: bad variable name. Game name on bnet where always the name of the player that started the game. I didn't rename it as it comes from multi.h so will do it once i get to there.

File: Source/automap.cpp [hunk 4]
```diff
@@ -68,20 +68,20 @@ void DrawSquare(const CelOutputBuffer &out, Point center, uint8_t color)
 
 void DrawMapVerticalDoor(const CelOutputBuffer &out, Point center)
 {
-	Point offest { center.x - AmLine32, center.y - AmLine16 };
+	Point offest { center.x - AmLine16, center.y - AmLine8 };
```
Comment (glebm) [inline Source/automap.cpp:71]:
```suggestion
	Point offset { center.x - AmLine16, center.y - AmLine8 };
```

File: Source/automap.cpp [hunk 5]
```diff
@@ -68,20 +68,20 @@ void DrawSquare(const CelOutputBuffer &out, Point center, uint8_t color)
...
-	DrawSquare(out, { center.x - AmLine16, center.y - AmLine8 }, MapColorsBright);
+	DrawLineTo(out, { center.x + AmLine16, offest.y }, { center.x + AmLine8, offest.y + AmLine4 }, MapColorsDim);
+	DrawLineTo(out, { offest.x, center.y + AmLine8 }, { offest.x + AmLine8, center.y + AmLine8 - AmLine4 }, MapColorsDim);
+	DrawSquare(out, center, MapColorsBright);
 }
 
 void DrawMapHorizontalDoor(const CelOutputBuffer &out, Point center)
 {
-	Point offest { center.x + AmLine32, center.y - AmLine16 };
+	Point offest { center.x + AmLine16, center.y - AmLine8 };
```
Comment (glebm) [inline Source/automap.cpp:80]: Same typo here

**Review by glebm — APPROVED** [review]

File: Source/automap.cpp (hunk 4 above)
Comment (AJenbo) [reply Source/automap.cpp]: I had a feeling this was coming :)

File: Source/automap.cpp (hunk 5 above)
Comment (AJenbo) [reply Source/automap.cpp]: https://github.com/diasurgical/devilutionX/compare/0aa653cd543e5f30873e1b3d6c8978baec229a55..fa371e62c902e72d1a708a75d8791ccd9c64c601

File: Source/automap.cpp [hunk 6]
```diff
@@ -68,20 +68,20 @@ void DrawSquare(const CelOutputBuffer &out, Point center, uint8_t color)
 
 void DrawMapVerticalDoor(const CelOutputBuffer &out, Point center)
 {
-	Point offest { center.x - AmLine32, center.y - AmLine16 };
+	const Point offest { center.x - AmLine16, center.y - AmLine8 };
```
Comment (glebm) [inline Source/automap.cpp:71]:
```suggestion
	const Point offset { center.x - AmLine16, center.y - AmLine8 };
```

here and below

**Review by glebm — APPROVED** [review]

**Pull request merged**